### PR TITLE
Externalize PV/PVC informer for StorageObjectInUseProtection & NodeAuthorizer

### PR DESCRIPTION
**Pull request by yue9944882**
**What this PR does / why we need it**:

/sig api-machinery

ref: #66680

We move on and make the change happen for PV/PVC. 

> PV: NodeAuthorizer, StorageObjectInUseProtection
> PVC: StorageObjectInUseProtection

1. Externalize PV and PVC informers for `StorageObjectInUseProtection`
2. Copy utility functions for PV from `pkg/api/persistentvolume` to `pkg/api/v1/persistentvolume` and make it accept external types.
3. Bump PV informer in NodeAuthorizer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/api/v1/persistentvolume/util.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	return ""
+}
+
+// Visitor is called with each object's namespace and name, and returns true if visiting should continue
+type Visitor func(namespace, name string, kubeletVisible bool) (shouldContinue bool)
+
+// VisitPVSecretNames invokes the visitor function with the name of every secret
+// referenced by the PV spec. If visitor returns false, visiting is short-circuited.
+// Returns true if visiting completed, false if visiting was short-circuited.
+func VisitPVSecretNames(pv *corev1.PersistentVolume, visitor Visitor) bool {
```
Comment (deads2k) [inline pkg/api/v1/persistentvolume/util.go]: Shouldn't this be paired with a delete of the existing internal one?

Comment (liggitt) [reply pkg/api/v1/persistentvolume/util.go]: if that's the only caller, yes

Comment (yue9944882) [reply pkg/api/v1/persistentvolume/util.go]: oh good point, cleaning

Comment (yue9944882) [reply pkg/api/v1/persistentvolume/util.go]: oops. internal ones are referenced by its REST 😅

[https://github.com/kubernetes/kubernetes/blob/91f93b74b17764fbf96bad1c26e8b53344169435/pkg/registry/core/persistentvolume/strategy.go#L56](https://github.com/kubernetes/kubernetes/blob/91f93b74b17764fbf96bad1c26e8b53344169435/pkg/registry/core/persistentvolume/strategy.go#L56)

**Comment by yue9944882** [comment]
@kubernetes/sig-testing-misc Report flaky test `TestVolumeBindingRescheduling` under `test/integration/scheduler/volume_binding_test`😃

File: pkg/api/v1/persistentvolume/util.go (hunk 1 above)
Comment (liggitt) [reply pkg/api/v1/persistentvolume/util.go]: can delete VisitPVSecretNames and associated tests from internal, right?

File: plugin/pkg/auth/authorizer/node/graph_populator.go [hunk 2]
```diff
@@ -212,7 +212,7 @@ func (g *graphPopulator) deleteVolumeAttachment(obj interface{}) {
 	if tombstone, ok := obj.(cache.DeletedFinalStateUnknown); ok {
 		obj = tombstone.Obj
 	}
-	attachment, ok := obj.(*api.PersistentVolume)
+	attachment, ok := obj.(*corev1.PersistentVolume)
```
Comment (liggitt) [inline plugin/pkg/auth/authorizer/node/graph_populator.go]: this is a pre-existing copy/paste bug... this should be `storagev1beta1.VolumeAttachment`

File: pkg/api/v1/persistentvolume/util.go (hunk 1 above)
Comment (yue9944882) [reply pkg/api/v1/persistentvolume/util.go]: oh.. i thought we were talking about the entire pv util pkg..

File: plugin/pkg/auth/authorizer/node/graph_populator.go (hunk 2 above)
Comment (liggitt) [reply plugin/pkg/auth/authorizer/node/graph_populator.go]: yeah, let me deal with that one separately. it will need picking back to older releases

Comment (yue9944882) [reply plugin/pkg/auth/authorizer/node/graph_populator.go]: will rebase onto that

**Comment by yue9944882** [comment]
/hold 
until #67471

File: pkg/api/v1/persistentvolume/util.go (hunk 1 above)
Comment (yue9944882) [reply pkg/api/v1/persistentvolume/util.go]: done

**Comment by deads2k** [comment]
please squash the bulk move together in one commit (not one for add and one for remove) and confirm it is a straight move.

Then please don't touch the same admission plugin in two commits. 

I think this is right.  I'll re-review after the commits are a little more logical.

**Comment by yue9944882** [comment]
> Then please don't touch the same admission plugin in two commits.

make sense. it's also pain when reviewing orderless commits

**Comment by deads2k** [comment]
/lgtm

nice progress.

**Pull request merged**